packages/hoc/src/withApollo.tsx

```diff
@@ -9,10 +9,8 @@ function getDisplayName<P>(WrappedComponent: React.ComponentType<P>) {
   return WrappedComponent.displayName || WrappedComponent.name || 'Component';
 }
 
-export function withApollo<TProps, TResult = any>(
-  WrappedComponent: React.ComponentType<
-    WithApolloClient<Omit<TProps, 'client'>>
-  >,
+export function withApollo<TProps extends WithApolloClient<{}>, TResult = any>(
```

Comment:
Wouldn't this fail if someone has given an explicit `TProps` argument to `withApollo` that excludes the `WithApolloClient` part to make the old version work. For example:

```
type Props = {user: string}
const ExampleC = ({user, client}: WithApolloClient<Props>) => {...}
const Example = withApollo<Props>(ExampleC)
```

Here, `Props` does not extend `WithApolloClient<{}>` and will cause a type error.